Apps/Src/SendDisplay.c

```diff
@@ -1,30 +1,296 @@
+/** Copyright (c) 2022 UT Longhorn Racing Solar
+ * @file SendDisplay.c
+ * @brief Function implementations for the display application.
+ * 
+ * This contains functions relevant to modifying states of specific
+ * components on our HMI design. The HMI has the ability to indicate 
+ * relevant information about system status to the driver.
+ * 
+ * @author Ishan Deshpande (IshDeshpa)
+ * @author Roie Gal (Cam0Cow)
+*/
 #include "SendDisplay.h"
 #include "Minions.h"
 #include "MotorController.h"
 #include <math.h>
 
-void Task_SendDisplay(void *p_arg) {
+/**
+ * Creates queue for display commands.
+ */
+#define DISP_Q_SIZE 5
+
+#define FIFO_TYPE Display_Cmd_t
+#define FIFO_SIZE sizeof(Display_Cmd_t)*DISP_Q_SIZE
+#define FIFO_NAME disp_fifo
+#include "fifo.h"
+
+disp_fifo_t msg_queue;
+
+static OS_SEM DisplayQ_Sem4;    // counting semaphore for queue message availability
+static OS_MUTEX DisplayQ_Mutex; // mutex to ensure thread safety when writing/reading to queue
+
+/**
+ * Enum and corresponding array for easy component selection.
+ */
+typedef enum{
+	// Boolean components
+	LEFT=0,
+	HEAD,
+	RIGHT,
+	HZD,
+	ARRAY,
+	MOTOR,
+	// Non-boolean components
+	VELOCITY,
+	ACCEL_METER,
+	SOC,
+	SUPP_BATT,
+	CRUISE_ST,
+	REGEN_ST,
+	GEAR,
+	// Fault code components
+	OS_CODE,
+	FAULT_CODE
+} Component_t;
+
+const char* compStrings[15]= {
+	// Boolean components
+	"ltime",
+	"head",
+	"rtime",
+	"hzd",
+	"arr",
+	"mot",
+	// Non-boolean components
+	"vel",
+	"accel",
+	"soc",
+	"supp",
+	"cruiseSt",
+	"rbsSt",
+	"gear",
+	// Fault code components
+	"oserr",
+	"faulterr"
+};
+
+/**
+ * @brief Pops the next display message from the queue and passes
+ * it to the display driver. Pends on semaphore and mutex to ensure that:
+ *  1) queue has messages to send (signaled by semaphore)
+ *  2) queue is not currently being written to by a separate thread (mutex)
+ * @returns ErrorStatus: ERROR or SUCCESS
+ */
+static ErrorStatus Display_PopNext(){
```

Comment:
Won't these be better in the display driver itself? SendDisplay can still use them if they're in the driver.

Reply:
It doesn't particularly matter, but it feels weird to have them in the Apps layer.

Reply:
It does feel kinda weird, but I was talking to @babusid and we came to the conclusion that Display Driver should pretty much just be core functionality of sending a display command and a few necessary commands (reset, fault, etc) and the Display App can do more complex stuff, since most of those functions are wrappers for the send command function anyways

Reply:
The problem with that approach is that now other tasks have to call functions from SendDisplay, instead of calling stuff from the driver. If everyone else thinks it's fine then we can keep it, but I'm still not 100% convinced.